fmriprep/interfaces/images.py

```diff
@@ -205,76 +191,148 @@ def _run_interface(self, runtime):
         all_shapes = np.array([img.shape for img in reoriented])
 
         # Identify images that would require excessive up-sampling
-        valid = self._prune_zooms(all_zooms, self.inputs.max_scale)
-        dropped_images = in_names[~valid]
+        valid = np.ones(all_zooms.shape[0], dtype=bool)
+        while valid.any():
+            target_zooms = all_zooms[valid].min(axis=0)
+            scales = all_zooms[valid] / target_zooms
+            if np.all(scales < self.inputs.max_scale):
+                break
+            valid[valid] ^= np.any(scales == scales.max(), axis=1)
 
         # Ignore dropped images
         valid_fnames = in_names[valid]
-        valid_imgs = orig_imgs[valid]
-        reoriented = reoriented[valid]
+        self._results['t1w_valid_list'] = valid_fnames.tolist()
 
         # Set target shape information
         target_zooms = all_zooms[valid].min(axis=0)
         target_shape = all_shapes[valid].max(axis=0)
+
+        self._results['target_zooms'] = tuple(target_zooms.tolist())
+        self._results['target_shape'] = tuple(target_shape.tolist())
+
+        # Create report
+        dropped_images = in_names[~valid]
+        segment = self._generate_segment(dropped_images, target_shape, target_zooms)
+        out_report = os.path.join(runtime.cwd, 'report.html')
+        with open(out_report, 'w') as fobj:
+            fobj.write(segment)
+
+        self._results['out_report'] = out_report
+
+        return runtime
+
+
+class ConformInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True, desc='Input T1w image')
+    target_zooms = traits.Tuple(traits.Float, traits.Float, traits.Float,
+                                desc='Target zoom information')
+    target_shape = traits.Tuple(traits.Int, traits.Int, traits.Int,
+                                desc='Target shape information')
+
+
+class ConformOutputSpec(TraitedSpec):
+    out_file = File(exists=True, desc='Conformed T1w image')
+
+
+class Conform(SimpleInterface):
+    """Conform a series of T1w images to enable merging.
+
+    Performs two basic functions:
+
+    #. Orient to RAS (left-right, posterior-anterior, inferior-superior)
+    #. Along each dimension, resample to minimum voxel size, maximum number of voxels
```

Comment:
Since this is no longer calculating the zooms, shape, we can just say:

"resample to target zooms (voxel sizes) and shape (number of voxels)"